packages/openid4vc-client/src/OpenId4VcClientService.ts

```diff
@@ -232,4 +294,89 @@ export class OpenId4VcClientService {
 
     return storedCredential
   }
+
+  public async requestCredential(agentContext: AgentContext, options: RequestCredentialOptions) {
```

Comment:
This method is quite similar to the request preAuthorized method. Do you think we could create a single method to request a credential that supports both flows (one has authorization code, other pre-authorized code), or do you think it would be good to keep them separate? If so it would be good to at least reduce duplication a bit to prevent the implementations from getting out of sync